bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Strand.java

```diff
@@ -115,7 +115,7 @@ public Strand(String name, StrandMetadata metadata, Scheduler scheduler, Strand
         if (currentTrxContext != null) {
             this.trxContexts = parent.trxContexts;
             this.trxContexts.push(currentTrxContext);
-            this.currentTrxContext = createTrxContextBranch(currentTrxContext, name);
+            this.currentTrxContext = currentTrxContext;
```

Comment:
Tests in https://github.com/ballerina-platform/module-ballerinai-transaction/pull/448